### ISSUE TITLE
Is there an empirically optimal maxVisits value that can yield more accurate winrate data?

### ISSUE THREAD

**Issue by qcgm1978** [lightvector/KataGo#833]
I would like to know if there is an empirically optimal maxVisits value that can yield more accurate winrate data. Setting a higher value requires more time, so I’m looking for a more economical way to obtain winrate fluctuation data across around 100 games. Are there any relevant studies on this? My current configuration is as follows:

···yml
path: "/opt/homebrew/Cellar/katago/1.13.2/bin/katago"
  arguments: "analysis -model /opt/homebrew/Cellar/katago/1.13.2/share/katago/g170-b30c320x2-s4824661760-d1229536699.bin.gz -config /opt/homebrew/Cellar/katago/1.13.2/share/katago/configs/analysis_example.cfg"
maxVisits: 16
```

**Comment by qcgm1978** [comment]
Thank you, I will look into the “figure of merit” you mentioned. I appreciate you providing this line of thinking - I will try to find a good solution, especially some kind of benchmark to validate the winrates.

**Comment by lightvector** [comment]
I would also recommend using a newer neural net. "g170-b30c320x2-s4824661760-d1229536699.bin.gz " is very old and likely much worse than the current 18b net at https://katagotraining.org/networks/ so if you want better analysis, you can probably get an improvement for free just by upgrading.

**Comment by qcgm1978** [comment]
Thanks. I've just download the highest Elo Rating Network File.

**Comment by kaorahi** [comment]
to get a rough idea...
![winrate](https://github.com/lightvector/KataGo/assets/38910552/405f4ba0-9c45-4cc8-bc27-8a3f6e3fd0b3)
![scoreLead](https://github.com/lightvector/KataGo/assets/38910552/fc2d639b-93b6-4065-85d8-4ed8d67b9ef2)

**Comment by qcgm1978** [comment]
I have another problem now. Which is more important for the accuracy of the win rate, maxVariationsForEachMove or maxVisits, and how to balance them to achieve higher efficiency?

**Comment by qcgm1978** [comment]
Google bard answer me as follows:

If you are playing in a tournament or other competitive setting, you should prioritize accuracy over efficiency. In this case, you should set maxVariationsForEachMove to a high value, such as 1000.

If you are playing for fun or practicing, you can prioritize efficiency over accuracy. In this case, you can reduce maxVariationsForEachMove to a lower value, such as 100 or 10.

If you are playing on a computer with limited resources, you will need to reduce both maxVariationsForEachMove and maxVisits to improve efficiency.

**Comment by lightvector** [comment]
"maxVariationsForEachMove" is not a parameter in KataGo, and I don't think the values given to you by bard are likely to have any meaningful use. You should set maxVisits as high as you can tolerate given your compute or time budget, and as far as I know there isn't any other parameter that it should be balanced against.

**Comment by qcgm1978** [comment]
@lightvector You're right. I misunderstood the option in [analyze-sgf](https://github.com/9beach/analyze-sgf/issues/15). In that js module maxVariationsForEachMove  have no effect on the accuracy. KataGo generates so many variations for a move, and the maximun variations count for a move of a SGF with the option.

**Comment by qcgm1978** [comment]
Based on my recent experiments, I've found that the optimal solution is likely to be found when the number of visits reaches around 2000.
